### PR TITLE
Virtual Thread ITs

### PR DESCRIPTION
**Pull request by cescoffier**
Verify the behavior of virtual threads

- RESTEasy Reactive (check dispatching strategy, filters, no pinning...)
- Reactive rest client (no pinning)
- gRPC client (no pinning)
- Mailer (both simple and template) (no pinning)
- Redis (client and cache) (no pinning)
- Enable native tests

### CODE REVIEW

File: integration-tests/virtual-threads/grpc-virtual-threads/pom.xml [hunk 1]
```diff
@@ -93,6 +93,42 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>${version.surefire.plugin}</version>
+                <configuration>
+                    <systemPropertyVariables>
+                        <junit.quarkus.orderer.secondary-orderer>
+                            org.junit.jupiter.api.ClassOrderer$OrderAnnotation
+                        </junit.quarkus.orderer.secondary-orderer>
+                    </systemPropertyVariables>
```
Comment (geoand) [inline integration-tests/virtual-threads/grpc-virtual-threads/pom.xml:120]: Is this really needed?

Comment (cescoffier) [reply integration-tests/virtual-threads/grpc-virtual-threads/pom.xml:120]: I don't believe it's necessary for these tests.

Comment (geoand) [reply integration-tests/virtual-threads/grpc-virtual-threads/pom.xml:120]: Yeah, I don't see it being of use iin the tests

File: .github/workflows/ci-actions-incremental.yml [hunk 2]
```diff
@@ -636,7 +636,7 @@ jobs:
           java-version: ${{ matrix.java.java-version }}
       - name: Run tests
         run: |
-          export LANG=en_US && ./mvnw -e -B -fae --settings .github/mvn-settings.xml -f integration-tests/virtual-threads clean verify -Dextra-args=${{matrix.java.extra-args}}
+          export LANG=en_US && ./mvnw -e -B -fae --settings .github/mvn-settings.xml -f integration-tests/virtual-threads clean verify -Dnative -Dextra-args=${{matrix.java.extra-args}} -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-20
```
Comment (geoand) [inline .github/workflows/ci-actions-incremental.yml:639]: We'll have to remember to update this

Comment (cescoffier) [reply .github/workflows/ci-actions-incremental.yml:639]: Yes, this is a temporary solution until 21 is there. 

When we add 21 support, we will pick the image automatically.

File: integration-tests/virtual-threads/grpc-virtual-threads/pom.xml (hunk 1 above)
Comment (cescoffier) [reply integration-tests/virtual-threads/grpc-virtual-threads/pom.xml]: I've cleaned up the pom files.

**Review by geoand — APPROVED** [review]

**Review by ozangunalp — APPROVED** [review]
This looks good. I have an upcoming branch on this adding the reactive messaging tests with virtual threads.

I was thinking, maybe we can bump Java 19 branch to Java 20 to include these tests in JVM.

That way, instead of filtering `integration-tests/virtual-threads` module out of the regular build we could only skip test execution <20. 
And for native tests, we could add a job, +/- in the same way as the current native tests to filter only impacted tests.

**Pull request merged**